File: R/2_even_fibo.r
even_fibo <- function(limit) {
  a <- 1
  b <- 1
  l <- NULL	

  while(a < limit) {
    if(a %% 2 == 0) { l <- c(l,a) } 
    c <- a + b
    a <- b
    b <- c
  }
  sum(l)
}

system.time( result <- even_fibo(4000000) )
result

File: ruby/2_even_fibo.rb
require "benchmark"

even_fibo = Enumerator.new do |y|
  a = b = 1
  loop do 
    y << a if(a % 2 == 0)
    a, b = b, a + b
  end
end

result = 0
time = Benchmark.realtime do
  result = even_fibo.take_while { |i| i < 4_000_000 }.inject(:+)
end
puts "Code using enumerator"
puts "#{result} in #{time * 1e6} microseconds"

